spring-boot/2d4e68a9777601bfb8309c94d8b74bc21be80ad1/original/README.md

```diff
@@ -26,7 +26,7 @@
     </plugin>
 ```
 
-## 5. Inside the folder  **spring-boot** run the command:
+## 5. Inside the folder **spring-boot/spring-boot** run the command:

```

Comment:
O mesmo problema comentado para o outro cenário se aplica aqui também.

`[ERROR]     'dependencies.dependency.version' for org.springframework:spring-context:jar is missing.`

Reply:
@galilasmb a possibilidade de replicação é extremamente importante para a gente. Podes, por favor, verificar as anotações de @leusonmario e tentar ajeitar e testar as modificações em um ambiente limpo e estável, de forma que não seja necessário retrabalho na análise e que a gente não corra o risco de poluir o repositório com cenários que levem a dificuldade de replicação?

Reply:
Fiz as correções, removendo os **
Talvez tenha funcionado em minha máquina normalmente porque eu já sabia os caminhos e não tenha aparecido corretamente por conta dos **

Reply:
O problema não é a remoção dos asteriscos (**). O problema é que versões de dependências não estão declaradas em arquivos POM. 

Inclusive, para não correr o risco de ser um problema apenas causado meu ambiente, pedi para Toni tentar gerar os jars, e ele reportou o mesmo problema.

Reply:
Você seguiu as orientações de @pauloborba  de testar as modificações em um ambiente limpo e estável? 
Talvez você consiga gerar os jars sem problemas porque seu ambiente já tem estes arquivos jars destas dependências salvas no seu ambiente.

Reply:
@galilasmb ?

Reply:
Baixei o projeto do zero e refiz todo o README, aqui funcionou normalmente. Não sei o que fazer para resolver teu problema.
Você alterou o POM do spring-boot-parent para https (passo 3)?
Ou conseguiu resolver isto?

Talvez seja que algumas dependências já estejam instaladas em minha máquina.

Reply:
Outra coisa que pode está acontecendo é com relação ao token do github para baixar dependências (única configuração que posso ter diferente da máquina de vocês), que eu já tenho adicionado na raiz em home na pasta .m2, o arquivo de nome settings.xml:

<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                      http://maven.apache.org/xsd/settings-1.0.0.xsd">

  <activeProfiles>
    <activeProfile>github</activeProfile>
  </activeProfiles>

  <profiles>
    <profile>
      <id>github</id>
      <repositories>
        <repository>
          <id>central</id>
          <url>https://repo1.maven.org/maven2</url>
          <releases><enabled>true</enabled></releases>
          <snapshots><enabled>true</enabled></snapshots>
        </repository>
        <repository>
          <id>spg</id>
          <name>spgroup</name>
          <url>https://maven.pkg.github.com/spgroup/soot</url>
        </repository>
	<repository>
          <id>svfa</id>
          <name>svfa repository</name>
          <url>https://maven.pkg.github.com/rbonifacio/svfa-scala</url>
        </repository>

      </repositories>
    </profile>
  </profiles>

  <servers>
    <server>
      <id>spg</id>
      <username>USERNAME-GITHUB</username>
      <password>TOKEN</password>
    </server>
     <server>
      <id>svfa</id>
      <username>USERNAME-GITHUB</username>
      <password>TOKEN</password>
    </server>
  </servers>
</settings>

Reply:
> Você alterou o POM do spring-boot-parent para https (passo 3)?

Sim, seguimos as instruções descritas neste passo.

> Talvez seja que algumas dependências já estejam instaladas em minha máquina.

Sim, isso foi justamente o que levantei na minha mensagem anterior.

> Outra coisa que pode está acontecendo é com relação ao token do github para baixar dependências.

Eu não acho que seja este o problema. 
Para garantir que o problema é a presença dos jars no seu ambiente, eu diria para você deletar os JARs desta dependência `org.springframework.boot:spring-boot-autoconfigure` do seu repositório local de dependências (m2). E depois, tentar gerar o build.

Reply:
Excluí todas as dependências locais, realmente ocorreu o erro de dependência, @leusonmario 

```
Non-resolvable import POM: Could not find artifact org.springframework:spring-framework-bom:pom:4.1.0.BUILD-SNAPSHOT in central (https://repo1.maven.org/maven2) @ org.springframework.boot:spring-boot-dependencies:1.2.0.BUILD-SNAPSHOT, /home/galileu/spring-boot/spring-boot-dependencies/pom.xml, line 1014, column 16
[ERROR] 'dependencies.dependency.version' for org.springframework:spring-context:jar is missing. @ line 27, column 15
[ERROR] 'dependencies.dependency.version' for org.springframework:spring-orm:jar is missing. @ line 142, column 15
[ERROR] 'dependencies.dependency.version' for org.springframework:spring-test:jar is missing. @ line 147, column 15
[ERROR] 'dependencies.dependency.version' for org.springframework:spring-web:jar is missing. @ line 152, column 15
[ERROR] 'dependencies.dependency.version' for org.springframework:spring-webmvc:jar is missing. @ line 173, column 15
```

Reply:
Como prosseguir? @pauloborba @leusonmario

Reply:
Diante do status atual, qual seria a melhor abordagem a tomar @pauloborba? @galilasmb continuaria a tentar resolver este problema das dependências?

Reply:
Eu diria para tentar resolver o problema destas dependências considerando que você já conseguiu gerar o jar em algum momento @galilasmb.

Reply:
sim @leusonmario , seria isso mesmo. @galilasmb, pelo que entendi  essas dependências estão especificadas no POM mas não estão sendo automaticamente baixadas por não estarem mais disponíveis no servidor. é isso? teria então que especificar um servidor que as contenha ou, no README, recomendar a inclusão manual dessas dependências, e disponibilizar os JARs como parte desse commit. seria isso @leusonmario ?

Reply:
Sim, é uma opção @pauloborba.
Apenas uma consideração sobre a causa o problema; acredito que este comportamento ocorre porque na declaração das dependências, não há a indicação da versão. Talvez a indicação das versões esteja presente em módulo, que não é "visível" do módulo que estamos tentando buildar; considerando que a indicação de gerar a build para apenas o módulo onde acontece mudanças em um mesmo arquivo. Se for isso mesmo, uma alternativa seria adicionar estas versões manualmente.

Reply:
Resolvido. Tive que buscar na mão uma versão das dependências que estivessem funcionando. Encontrei e alterei manualmente, adicionando no arquivo POM das dependências. Creio que em minha máquina não houve esse problema, porque já executei vários outros cenários deste projeto e já havia baixado estas dependências ou uma mais recente.